tests/v2/validation/provisioning/rke2/etcd_bkp_restore_test.go

```diff
@@ -0,0 +1,378 @@
+	r.watchAndWaitForPods(client, clusterID)
+	logrus.Infof("All pods are up and running.............")
+
+	// creating the workload W1
+	logrus.Infof("creating a workload(nginx deployment).............")
+
+	labels := map[string]string{}
+	labels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", r.ns, w1Name)
+
+	containerTemplate := workloads.NewContainer("ngnix", "nginx", v1.PullAlways, []v1.VolumeMount{}, []v1.EnvFromSource{})
+	podTemplate := workloads.NewPodTemplate([]v1.Container{containerTemplate}, []v1.Volume{}, []v1.LocalObjectReference{}, labels)
+	deployment := workloads.NewDeploymentTemplate(w1Name, r.ns, podTemplate, labels)
+
+	// creating steve client
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	require.NoError(r.T(), err)
+
+	deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).Create(deployment)
+	require.NoError(r.T(), err)
+
+	require.Equal(r.T(), w1Name, deploymentResp.ObjectMeta.Name)
+	logrus.Infof("created a workload(nginx deployment).............")
+
+	logrus.Infof("creating watch over w1(nginx-deployment).............")
+	r.watchAndWaitForDeployment(client, clusterID, w1Name)
+	logrus.Infof("w1(nginx-deployment) is ready.............")
+
+	// creating the ingress1
+	logrus.Infof("creating an ingress.............")
+
+	exactPath := networkingv1.PathTypeExact
+	paths := []networkingv1.HTTPIngressPath{
+		{
+			Path:     "/index.html",
+			PathType: &exactPath,
+			Backend: networkingv1.IngressBackend{
+				Service: &networkingv1.IngressServiceBackend{
+					Name: w1ServiceName,
+					Port: networkingv1.ServiceBackendPort{
+						Number: 80,
+					},
+				},
+			},
+		},
+	}
+	ingress := ingresses.NewIngressTemplate(ingress1Name, r.ns, "", paths)
+
+	ingress1, err := steveclient.SteveType(ingresses.IngressSteveType).Create(ingress)
+	require.NoError(r.T(), err)
+
+	require.Equal(r.T(), ingress1Name, ingress1.ObjectMeta.Name)
+	logrus.Infof("created an ingress.............")
+
+	logrus.Infof("creating a snapshot of the cluster.............")
+	err = createSnapshot(client, clusterName, 1, r.ns)
+	require.NoError(r.T(), err)
+	logrus.Infof("created a snapshot of the cluster.............")
+
+	logrus.Infof("creating watch over cluster after creating a snapshot.............")
+	r.watchAndWaitForCluster(kubeProvisioningClient, clusterName)
+	logrus.Infof("cluster is active again.............")
+
+	var snapshotToBeRestored string
+
+	err = kwait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		snapshotList, err := getSnapshots(client, localClusterID)
+		if err != nil {
+			return false, err
+		}
+		totalClusterSnapShots := 0
+		for _, snapshot := range snapshotList {
+			if strings.Contains(snapshot.ObjectMeta.Name, clusterName) {
+				if snapshotToBeRestored == "" {
+					snapshotToBeRestored = snapshot.Name
+				}
+				totalClusterSnapShots++
+			}
+		}
+		if totalClusterSnapShots == 3 {
```

Comment:
why is it an error if there are 3 snapshots? Is there a way to make this parameter based instead, like based on the snapshotList length?